### PR TITLE
Kill json_response

### PR DESCRIPTION
**Pull request by orangejenny**
See https://github.com/dimagi/commcare-hq/pull/24069

### CODE REVIEW

**Review by nickpell — APPROVED** [review]
👍

File: corehq/apps/reports/generic.py [hunk 1]
```diff
@@ -674,12 +674,12 @@ def filters_response(self):
 
     @property
     @request_cache()
-    def json_response(self):
+    def JsonResponse(self):
```
Comment (dannyroberts) [inline corehq/apps/reports/generic.py]: I think this change must have been unintentional, right?

**Review by dannyroberts — COMMENTED** [review]
🎉 thanks for taking the plunge for us @orangejenny!

File: corehq/apps/reports/generic.py (hunk 1 above)
Comment (dannyroberts) [reply corehq/apps/reports/generic.py]: As per usual @emord seems to have beaten me to it: https://github.com/dimagi/commcare-hq/pull/24090

**Comment by dannyroberts** [comment]
@nickpell It looks like `JsonResponse` isn't a drop-in replacement for `json_reponse` after all. I think unless someone knows exactly what to do with this, it seems like we should just revert this PR and possibly un-deprecate `json_response`

**Comment by emord** [comment]
@dannyroberts 

> we should just revert this PR 

+1

> possibly un-deprecate json_response

I think deprecation is still correct. We want future code to not use `json_response` and use more portable django code

**Comment by dannyroberts** [comment]
I'm thinking if we find that our code assumes a certain set of presets, then we could just make json_response a wrapper for `JsonResponse` with those presets. I could see it going either way whether that's a good permanent state. I'll start by just reverting this PR

**Pull request merged**